jobs/build-charms/charms.py

```diff
@@ -387,7 +387,7 @@ def _read_metadata_resources(self):
     def setup(self):
         """Setup directory for charm build"""
         downstream = f"https://github.com/{self.opts['downstream']}"
```

Comment:
is downstream guaranteed to be set here?

Reply:
TLDR: yes guaranteed.

TL part:
So this code has a lot of problems.  But `def setup(...)` is only called by the charm build job and not the bundle build-job.  The charm build uses this list [charm-support-matrix.inc](https://github.com/charmed-kubernetes/jenkins/blob/master/jobs/includes/charm-support-matrix.inc).  
I have another MAJOR refactor (https://github.com/charmed-kubernetes/jenkins/pull/744) in this repo which addresses some of these oddities